src/hooks/useCodapData.ts

```diff
@@ -104,10 +105,43 @@ export const useCodapData = () => {
     }
   };
 
+  const extractUniqueLocations = (allItems: any): ILocation[] => {
+    const uniqueLocations: ILocation[] = [];
+
+    allItems.forEach((c: any) => {
+      const locationObj: ILocation = {
+        name: c.values.location,
+        latitude: c.values.latitude,
+        longitude: c.values.longitude,
+        coordinatePair: `(${c.values.latitude},${c.values.longitude})`
```

Comment:
Is the idea here to have something like a unique ID/key that can be easily compared? It's a pretty cool idea. However, I think I'll need a helper function like `locationsEqual = (a, b) => a.lat === b.lat && a.long === b.long` to check if a given lat/long matches any option in the dropdown. Then we could probably skip storing this derived property. This equality helper should probably be universal throughout the whole app. Alternatively, another option could be to generate this ID/key from the lat/long pair and compare the result. But it's pretty minor dilemma. 😉